src/encoding/draw.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Also licensed under MIT license, at your choice.
+
+use bytemuck::{Pod, Zeroable};
+use peniko::{BlendMode, Color};
+
+use super::Monoid;
+
+/// Draw tag representation.
+#[derive(Copy, Clone, PartialEq, Eq, Pod, Zeroable)]
+#[repr(C)]
+pub struct DrawTag(pub u32);
+
+impl DrawTag {
+    /// No operation.
+    pub const NOP: Self = Self(0);
+
+    /// Color fill.
+    pub const COLOR: Self = Self(0x44);
+
+    /// Linear gradient fill.
+    pub const LINEAR_GRADIENT: Self = Self(0x114);
+
+    /// Radial gradient fill.
+    pub const RADIAL_GRADIENT: Self = Self(0x2dc);
+
+    /// Image fill.
+    pub const IMAGE: Self = Self(0x48);
+
+    /// Begin layer/clip.
+    pub const BEGIN_CLIP: Self = Self(0x9);
+
+    /// End layer/clip.
+    pub const END_CLIP: Self = Self(0x21);
+}
+
+impl DrawTag {
+    /// Returns the size of the info buffer used by this tag.
+    pub fn info_size(self) -> u32 {
+        (self.0 >> 6) & 0xf
+    }
+}
+
+/// Draw data for a solid color.
+#[derive(Clone, Copy, Debug, Default, Zeroable, Pod)]
+#[repr(C)]
+pub struct DrawColor {
+    /// Packed RGBA color.
```

Comment:
This would probably be a good place to document the byte order.

Reply:
Ah, good point since it’s technically ABGR and also premultiplied.